src/main/java/net/openhft/chronicle/bytes/internal/HeapBytesStore.java

```diff
@@ -538,15 +540,24 @@ public void write(
                         this.dataOffset + offsetInRDO, length);
 
             } else {
-                memory.copyMemory(bytes.array(), offset, realUnderlyingObject,
-                        this.dataOffset + offsetInRDO, length);
+                byte[] src = bytes.array();
+
+                copyMemory0(offsetInRDO, offset, length, src);
             }
         } catch (NullPointerException ifReleased) {
             throwExceptionIfReleased();
             throw ifReleased;
         }
     }
 
+    private void copyMemory0(long offsetInRDO, int offset, int length, byte[] src) {
```

Comment:
is this behaviour change?

Reply:
This is an inlining of a removed method in UnsafeMemory.